ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,486 @@
+    interpolated_htsgw_surface.attrs['units'] = 'm'
+    interpolated_wind_surface = xr.DataArray(interpolated_wind_surface_values, coords={'time': satellite_time_unix}, dims=['time'], name='WIND_surface_interpolated')
+    interpolated_wind_surface.attrs['units'] = 'm/s'
+    longitude_dataarray = xr.DataArray(filtered_satellite_longitudes, coords={'time': satellite_time_unix}, dims=['time'], name='longitude')
+    longitude_dataarray.attrs['units'] = 'degrees_east'
+    latitude_dataarray = xr.DataArray(filtered_satellite_latitudes, coords={'time': satellite_time_unix}, dims=['time'], name='latitude')
+    latitude_dataarray.attrs['units'] = 'degrees_north'
+    hs_cal_dataarray = xr.DataArray(hs_cal.values, coords={'time': satellite_time_unix}, dims=['time'], name='hs_cal_time_averaged_satellite')
+    hs_cal_dataarray.attrs['units'] = 'm'
+    wsp_cal_dataarray = xr.DataArray(wsp_cal.values, coords={'time': satellite_time_unix}, dims=['time'], name='wsp_cal_time_averaged_satellite')
+    wsp_dataarray.attrs['units'] = 'm/s'
+   # fcst_hr_dataarray = xr.DataArray(interpolated_wind_surface_times, coords={'time': satellite_times_unix}, dims=['time'], name='fcst_time').assign_attrs(units='seconds') #fcst_hr
+   # interpolated_dataset['fcst_hr2_swh'] = xr.DataArray(fcst_hr2_ws, dims=['time'], name='fcst_hr2_ws').assign_attrs(description="Forecast hour relative to initial condition time for significant wave height", units='hours')
+    fcst_hr_dataarray = xr.DataArray(fcst_hr2_ws, coords={'time': satellite_times_unix}, dims=['time'], name='fcst_time').assign_attrs(units='hours') #fcst_hr
+    
+
+    # Combine into a single dataset / 'time': satellite_time_unix
+    interpolated_dataset = xr.Dataset({
+        'time': time_dataarray,
+        'HTSGW_surface_interpolated': interpolated_htsgw_surface,
+        'WIND_surface_interpolated': interpolated_wind_surface,
+        'hs_time_averaged_satellite': hs_dataarray,
+        'wsp_time_averaged_satellite': wsp_dataarray,
+        'longitude': longitude_dataarray,
+        'latitude': latitude_dataarray,
+        'hs_cal_time_averaged_satellite':hs_cal_dataarray,
+        'wsp_cal_time_averaged_satellite':wsp_dataarray,
+        'fcst_hr': fcst_hr_dataarray
+    })
+
+    # Before saving your dataset, add the reference times as attributes
+    interpolated_dataset.attrs['satellite_name'] = satellite_name
+    interpolated_dataset.attrs['reference_time'] = reference_time
+    interpolated_dataset.attrs['reference_time'] = f"{reference_time} seconds since 1970-01-01 00:00:00"
+    base_output_filename = "ProcessedData"  
+    interpolated_dataset.attrs['model_name'] = model_name
+
+    # Save the combined dataset to a NetCDF file
+    output_filename = f"{base_output_filename}_{satellite_name}_{reference_date}.nc"  # Adjusted to use reference_date
+    interpolated_dataset.to_netcdf(output_file, format='NETCDF4')
+
+
+
+
+def main():
+    # Command-line arguments
```

Comment:
@Ghazal-Mohammadpour  I saw that you merged in https://github.com/NOAA-EMC/WW3-tools/pull/46 but then have since reverted all of the updates from that PR, I'm curious if that was intentional?